### ISSUE TITLE
Need way to expand/search guild members list - can't assign new leader or boot member when member isn't listed

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#1806]
At present, Guilds only list the first 15 members, plus a little blob saying how many more members than that you have. But there's not any way to access the rest of that list--today I wanted to know if somebody was still a guild member, and had no means of determining that.

Suggestions, in order of my most preferable to least (though you could do more than one):
1. Add a search field to the member list, something like what we have in the Public Guilds tab.
2. Add a button to click to load the full list of guild members.
3. Implement incremental-load scrolling, so you can roll down to the bottom of the list, it picks up the next 15 (or however many) names, and you continue...
##

**Comment by SabreCat** [comment]
Related: #2160. I'm not sure you can boot a problem guildie who doesn't show up on the list, either.

**Comment by wc8** [comment]
This is a problem for me, too. 
If this is intended to be a sampling of members it should be clearer, at least, but I think there are many reasons we need full member lists.

Also, the round # more members label at the bottom looks like it's a button. More than a few times I've found myself trying to click it repeatedly until I remember. If I'm forced to remind myself that doesn't work that way, probably other users are thinking something's broken. Is it really supposed to not do anything?

**Comment by brightbold** [comment]
I, too, would really like this feature, if for no other reason because sometimes before I post something in a guild I want to be sure my kid is not a member! And in conjunction with [#2239], even if it's a small guild and there are only a few members not showing, there's no way to see them. 

Like @wc8 I keep thinking I should be able to click on "X more members" and get the full list. But I'm not really picky about how it's implemented - do whatever's best, performance-wise, although if we go for an infinite scroll type thing, we should say the number of members up at the top of the list, so you don't start trying to scroll to the bottom without realizing the guild has 3000 members.

**Comment by habitrpguser** [comment]
One year later, still the same problem. I tried via https://habitrpg.com/static/api , and that only gave the first-15-members list, too! Also, I can't find a way in the API site either to reassign a guild leader, even if I already know their UID and just assume they're a guild member. _(Update: Ok, I overlooked the guild editing feature via POST /groups/{gid}, but I think the risk of accidentally destroying the guild by posting the schema wrong is too high to try it.)_ It's an acute problem at the moment because, in [this guild](https://habitrpg.com/#/options/groups/guilds/bb511491-561b-4f36-95ff-0c8d8e6d8660), I started a challenge for guild leadership, said to end today, and the winner (UID f25852d5-fe6b-43e6-bb9b-f1cc304547f2) is supposed to become guild leader. Also, it's planned (though, ultimately, up to the people who become guild leader) to make similar challenges in the future. Which, at the moment, just isn't possible in guilds with more than 15 members. (I'm currently very tempted to just kick out enough people, send them messages that it was necessary for technical reasons and that they are welcome to rejoin, then transfer guild leadership. Yes, I'm mean like that.)

**Comment by deilann** [comment]
Or you can just ask to have that person set as the guild leader. Give us
the group ID and the user's UUID.

**Comment by habitrpguser** [comment]
@deilann Oooh, that was quick, thanks! GID is bb511491-561b-4f36-95ff-0c8d8e6d8660 (as seen from the link I gave), UID of the winner and hopefully-soon-guild-leader is f25852d5-fe6b-43e6-bb9b-f1cc304547f2 . I just officially finished the challenge and declared her winner.

**Comment by deilann** [comment]
@Alys can you assign this?

**Comment by habitrpguser** [comment]
... and I was so looking forward to finally getting to keelhaul someone... ;> - Btw. it would only be 5 people, to whom I would send friendly messages with links to the guild and technical rejoining instructions + wiki link, and none of them has ever said anything in guild chat, or shown other signs of guild activity. So it wouldn't be a tragedy. But it would be helpful to debug this or give a workaround for future similar challenges!

**Comment by Alys** [comment]
The guild leader has been changed.

**Comment by habitrpguser** [comment]
@Alys Wow, thanks for the quick solution! So glad to have admins like you all - kudos! :bow:

**Comment by habitrpguser** [comment]
Just a few new data about trying to (as a workaround) assign a new guild leader via https://habitrpg.com/static/api : Well, I tried, it didn't work, I talked to @Alys about it, she has some ideas what could be the reasons, and I'll just copy part of the posts here for details:

me:
"My evil evil plan was to use this guild to experiment around with making someone guild leader via Swagger UI. Tried it two times now (once with sending the full guild scheme, once with just the id and new leader id, and got  `"err": "Group not found"` both times. I'll keep testing!"

@Alys :
"I suspect that the POST /groups/{gid} (Edit a group) section in the API interface hasn't been set up correctly. There should probably be a field for entering the GID. It's also possible that it should be PUT, not POST. Cf the user item for editing a task. If you'd care to log a bug report, we'd get that investigated properly for you. Eventually. :("

Since I'm not really sure if it is a bug or if it's just me, I thought I'd add it here. If it's a separate bug, you can still open it as a new issue.

**Comment by BloodyJennyFlint** [comment]
Same problem, same group- The winner of the become the guild leader challenge needs to be made guild leader, but is not in the 15 members shown on the drop-down. 
Group ID  bb511491-561b-4f36-95ff-0c8d8e6d8660
User ID for new guild leader b314f798-b561-4b80-a68b-b25da898cdb0
Help, please and thank you!

**Comment by Alys** [comment]
@BloodyJennyFlint I've changed the guild leader. Let me know if there's any problems!

**Comment by crookedneighbor** [comment]
@BloodyJennyFlint How long is the next challenge? I want to add a way for the leader to do this before the next one is due.

**Comment by BloodyJennyFlint** [comment]
Thanks @alys.  @crookedneighbor I'm not sure, since the new guild leader will be setting it up. Probably at least a couple weeks.

**Comment by Alys** [comment]
There's some discussion about the technical challenges of this at https://github.com/HabitRPG/habitrpg/pull/4889#issuecomment-89956261 and the comment(s) below it.

I am wondering if it would be worth having a full members list that shows only the User ID. That could be obtained from the group's own document and no additional database queries would be needed. It would be terribly ugly and potentially confusing, but it would give a guild/party leader the ability to assign any member as a new leader or boot any member. Because of the ugliness, I'd suggest that this list was available to only the group leader, and there'd need to be an explanation at the top about why names weren't shown. The leader could click on any ID to see details for that user in a modal, and the modal would have options for booting or making the player the new leader. Those options would not exist in the list itself to ensure that the leader did check the person's name before taking an action on the person. Typically when you want to boot or promote a member, you would already be able to find their User ID (e.g., from their name in a chat post), so you could use your browser's find feature to search through the full list for it. Opinions welcome!

**Comment by habitrpguser** [comment]
Sounds good to me!

**Comment by Alys** [comment]
Are there any objections to my idea above about having a UUID-only list of members available to the group owner? If not, I'll mark this as suitable for someone to work on it. It would be a temporary feature to allow leadership changes and booting, until we work out how we want to handle this permanently, which might take us a while.

**Comment by crookedneighbor** [comment]
I don't think we should work in this right now, but we should consider it for v3.

**Comment by melissamcewen** [comment]
Can someone add a70c8dd3-754c-472e-9aad-29f1bafe2273
 as a leader of e02f5aae-f406-41f8-a24f-b2a85316cc0e? Or is there a way to do this with the API? Thanks!

**Comment by Alys** [comment]
@melissamcewen I can, but I need to be certain that it's the current guild leader who's requesting it and that's most easily done if you request it through Habitica itself, where I can confirm your User ID from your profile. Sorry about that! It's just standard practice we follow whenever we change a group's leader. Could you please copy your message and send it to me as PM or paste it into the Newbies guild and mention my name? 

This is not to say you did anything wrong by asking here! This is just one specific instance where Habitica messages happen to have an advantage. :)

**Comment by melissamcewen** [comment]
@Alys thanks I sent a message just now

**Comment by Alys** [comment]
@melissamcewen Thank you! I've changed the guild leader.

**Comment by paglias** [comment]
Fixed at API level but UI is still missing the ability to fetch other members

**Comment by Melissarie** [comment]
I would also like to see more of our party members. I have been looking everywhere trying to find where to put this. We have an old, large party and there are a lot of members, but we can mostly only see the inactive players which is causing a real issue with our current active members. We can't even see who all is participating in a quest, and it's causing a few problems. I'm attempting to add a screenshot of one of the things I'm referring to (The other half of my member pictures at the top are in the inn it seems, so all I could do was have it show the last people logging in before the others)
![members](https://cloud.githubusercontent.com/assets/19338005/16416420/c1d72470-3d07-11e6-8982-1921c2b57dac.png)

**Comment by paglias** [comment]
We can decide to fetch all party members but IMHO only after we enable the
cap to the number of members (old big parties will be grandfathered)
Il 28 giu 2016 3:10 PM, "Melissarie" notifications@github.com ha scritto:

**Comment by Alys** [comment]
@Melissarie I'm not sure if you saw my answer to you when you asked about that in the Report a Bug guild: "It's not possible to affect which party members are shown. However if your party leader wishes it, I can help them remove all members who haven't logged in to Habitica for the past several days/weeks (the leader can choose the period of time). The leader can post here [to the Report a Bug guild; not in GitHub] if they want me to do that. I recommend it because it will help your party be more manageable and should make the party page and website header load more quickly."

**Comment by khipkin** [comment]
What if a new sort option were added where you could sort party members based on how recently they were active?

**Comment by Melissarie** [comment]
@Alys I didn't see that, thank you. I'll have to talk to the party leader

**Comment by sebcagnon** [comment]
Hello

At least for the Guild Leader part, I propose that we change the dropdown by text box where we can input the ID of the new Guild Leader. The request should error if the ID is wrong or the user is not in the guild.

**Comment by Alys** [comment]
I'm putting this on hold because we're most of the way through a project to change the website's front-end code and this might no longer happen afterwards, or if it does, there'll be new code that will need to be modified to fix it.

**Comment by lemoness** [comment]
I'm closing this for now because we have rewritten all of our front-end code, but if this is still an issue, we can open a new ticket.
